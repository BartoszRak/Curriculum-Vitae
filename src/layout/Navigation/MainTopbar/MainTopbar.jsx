import React from 'react'
import PropTypes from 'prop-types'
import {
  Toolbar, Typography, useTheme, useMediaQuery,
} from '@material-ui/core'
import { injectIntl } from 'react-intl'

import LanguageSelect from '~components/LanguageSelect'
import ResponsiveMenu from '~components/ResponsiveMenu'

import useStyle from './MainTopbar.style'

export function MainTopbar({ classes: overridingClasses, intl }) {
  const classes = { ...useStyle(), ...overridingClasses }
  const { formatMessage } = intl
  const theme = useTheme()
  const matchDimensions = useMediaQuery(theme.breakpoints.down('md'))

  const items = [
    {
      name: formatMessage({ id: 'layout.navigation.mainTopbar.items.me' }),
      props: {
        href: '/',
      },
    },
    {
      name: formatMessage({ id: 'layout.navigation.mainTopbar.items.offer' }),
      props: {
        href: '/offer',
      },
    },
  ]

  return (
    <Toolbar className={classes.root}>
      {!matchDimensions && (
        <Typography className={classes.title} variant="h1">
          Bartosz Rak
        </Typography>
      )}
      <ResponsiveMenu className={classes.menu} items={items} />
      <div>
        <LanguageSelect />
      </div>
    </Toolbar>
  )
}

MainTopbar.propTypes = {
  classes: PropTypes.object,
  intl: PropTypes.object,
}

export default injectIntl(MainTopbar)
