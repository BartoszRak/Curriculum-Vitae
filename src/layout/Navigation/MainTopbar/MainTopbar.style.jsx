import { makeStyles } from '@material-ui/core'

export default makeStyles(({ palette }) => ({
  root: {
    background: palette.primary.dark,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    fontSize: '1.9rem',
    letterSpacing: 2,
    textTransform: 'uppercase',
  },
}))
