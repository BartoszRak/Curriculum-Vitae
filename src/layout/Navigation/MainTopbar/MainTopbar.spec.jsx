import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { MainTopbar } from './MainTopbar'

jest.unmock('@material-ui/core')

jest.mock('react-router-dom', () => ({
  matchPath: jest.fn(),
  withRouter: jest.fn(),
}))
jest.mock('~components/LanguageSelect', () => 'LanguageSelectMock')

let props

describe('component: MainTopbar', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      intl: {
        formatMessage: jest.fn(),
      },
      location: {
        pathname: 'mockedPathname',
      },
    }
  })
  describe('rendering', () => {
    test('render without crush', () => {
      const wrapper = shallow(<MainTopbar {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = shallow(<MainTopbar {...props} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render language select', () => {
      const wrapper = shallow(<MainTopbar {...props} />)

      expect(wrapper.find('LanguageSelectMock').length).toBe(1)
    })
  })
})
