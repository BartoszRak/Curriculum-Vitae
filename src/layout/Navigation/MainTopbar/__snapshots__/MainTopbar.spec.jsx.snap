// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: MainTopbar rendering match snapshot  1`] = `
<WithStyles(ForwardRef(Toolbar))
  className="makeStyles-root-1"
>
  <div>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      variant="h1"
    >
      Bartosz Rak
    </WithStyles(ForwardRef(Typography))>
  </div>
  <div>
    <LanguageSelectMock />
  </div>
</WithStyles(ForwardRef(Toolbar))>
`;
